### ISSUE TITLE
SwatchColorPicker Keyboard Navigation doesn't work as expected

### ISSUE THREAD

**Issue by sivapalaniappan** [microsoft/fluentui#19175]
### Environment Information

- **Package version(s)**: 8.23.10
- **Browser and OS versions**: Edge - 92.0.902.55 (Official build) (64-bit) and Win10 Enterprise 21H1 (OS Build: 19043.1110)

### Describe the issue:

Incorrect Keyboard navigation for color selection on SwatchColorPicker.

### Please provide a reproduction of the issue in a codepen:

1. The repro happens with this fluent example itself https://fabricweb.z5.web.core.windows.net/oufr/8.23.10/#/examples/swatchcolorpicker
2. On the multiple rows example, Click any one of the colors on first 2 rows which is also not at the end of that row.
3. Navigate to end of that row using the Right Arrow Key and press Down Arrow Key until you reach the last color of the last row.
4. Now press Down Arrow Key again to see how selection shifts.

#### Actual behavior:

The selection doesn't go to the same column in the destination row but rather appears to be going to the color that was previously selected while we previously navigated in the destination row.

#### Expected behavior:

The selection should shift within the same column where applicable which is not happening even when we have a symmetrical grid.
